### DISCUSSION TITLE
Clarity on REST API guidelines

### DISCUSSION THREAD

**Discussion by alechenninger** [keycloak/keycloak#12701] [Ideas]
[Keycloak API guidelines](https://github.com/keycloak/keycloak-community/blob/main/design/rest-api-guideline.md) are draft, but we try to exercise them anyway as a starting point. 

Referring to API groups in a URL path, these say "Group names should be in lowercase and a single word name." but as far as I can tell, keycloak admin API uses dashes to separate words. Should a multi-word API group it be "word-word" or "wordword" or "word_word"?

Happy to submit a PR to clarify a stance here if there is some consensus.

**Comment by abstractj** [comment]
@alechenninger an update to the REST API guidelines would be more than welcome. @pedroigor did a stellar job working on it.

From my understanding, it is considered best practice to follow the approach using hyphens. That means "word-word". My reference is https://restfulapi.net/resource-naming/ which contains a lot of best practices for REST endpoints. Based on the section 2.2.3 hyphens are recommended to improve readability and underscores avoided.

**Reply by pedroigor** [reply]
@abstractj @alechenninger It makes sense to make this point clear. I agree with @abstractj that it should be using a hyphen. It should also be aligned with OpenBanking API guidelines (or at least what I see from specs and examples).

**Reply by alechenninger** [reply]
PR for clarification: https://github.com/keycloak/keycloak-community/pull/343
